contracts/NodeVersionBeacon.cdc

```diff
@@ -336,6 +350,17 @@ pub contract NodeVersionBeacon {
             // If we passed a boundary re-emit the VersionBeacon event
             NodeVersionBeacon.emitEventOnNextHeartbeat = true
         }
+
+        /// Emit a ProtocolStateVersionUpgrade event, if a pending upgrade is in storage.
+        access(self) fun emitProtocolStateVersionUpgradeEventIfNeeded() {
+            let pendingUpgrade = NodeVersionBeacon.popPendingProtocolStateVersionUpgrade()
+            if pendingUpgrade == nil {
+                return
+            }
+            emit NodeVersionBeacon.ProtocolStateVersionUpgrade(
+                newProtocolVersion: pendingUpgrade!.version,
```

Comment:
It doesn't look like it from looking at the other code, but I just want to confirm that there is no way that this optional can be `nil` here

Reply:
I'm pretty sure it cannot be nil, because of the `if` statement above:
```cadence
if pendingUpgrade == nil {
    return
}
```